akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -166,7 +174,10 @@ private[akka] class TLSActor(settings: ActorMaterializerSettings,
       case Some(TLSClientAuth.Need) ⇒ engine.setNeedClientAuth(true)
       case _                        ⇒ // do nothing
     }
-    params.sslParameters foreach (p ⇒ engine.setSSLParameters(p))
+    params.sslParameters foreach { p ⇒
+      hostInfo foreach { case (host, _) ⇒ applySNI(host, engine, p) }
```

Comment:
so you propose to move `applySNI` to ssl-config—I’m not sure whether that’s worth it, because we cannot make people remember to apply it